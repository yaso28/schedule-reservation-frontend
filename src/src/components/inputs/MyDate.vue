<template>
  <MyFormGroup
    :label="label"
    :form="form"
    :field="field"
    :label-for="id"
  >
    <MyDateNoWrapper
      :form="form"
      :field="field"
      :show-clear="showClear"
    />
  </MyFormGroup>
</template>

<script>
export default {
  name: 'MyDate',
  components: {
    MyFormGroup: require('./MyFormGroup').default,
    MyDateNoWrapper: require('./MyDateNoWrapper').default
  },
  mixins: [require('./input-mixin').default],
  props: {
    showClear: {
      type: Boolean,
      default: true
    }
  }
}
</script>
