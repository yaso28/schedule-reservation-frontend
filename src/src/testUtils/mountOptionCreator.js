import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
