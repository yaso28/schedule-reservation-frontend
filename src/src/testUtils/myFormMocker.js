import MyForm from '@/models/MyForm'
jest.mock('@/models/MyForm')

class MyFormMocker {
  createMockSingleField (field, initValue) {
    const input = {}
    input[field] = initValue
    return this.createMock(input)
  }

  createMock (input) {
    const mockForm = new MyForm(input)
    mockForm.values = input
    mockForm.errors = {}
    return mockForm
  }

  async setValue (wrapper, form, field, value) {
    await wrapper.setProps({
      form: Object.assign({}, form, {
        values: Object.assign({}, form.values, { [field]: value })
      })
    })
  }

  setValueNonProps (form, field, value) {
    form.values[field] = value
  }

  async setError (wrapper, form, field, errorMessage) {
    await wrapper.setProps({
      form: Object.assign({}, form, {
        errors: Object.assign({}, form.errors, { [field]: errorMessage })
      })
    })
  }
}

export default new MyFormMocker()
