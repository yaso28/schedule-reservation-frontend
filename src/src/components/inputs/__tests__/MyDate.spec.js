import { mount } from '@vue/test-utils'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import MyDate from '../MyDate'
import myFormMocker from '@/testUtils/myFormMocker'

jest.mock('@/models/MyForm')

function getInputWrapper (wrapper) {
  return wrapper.find('input')
}
function getButtonWrapper (wrapper) {
  return wrapper.find('button')
}

describe('MyDate', () => {
  let form
  let field
  let wrapper
  beforeEach(() => {
    const label = 'サンプル'
    field = 'sample_field'
    form = myFormMocker.createMockSingleField(field, '')

    const mountOption = mountOptionCreator.create()
    mountOption.propsData = { label, form, field }
    wrapper = mount(MyDate, mountOption)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  describe('通常', () => {
    test('表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('バリデーションエラー', () => {
    beforeEach(async () => {
      await myFormMocker.setError(wrapper, form, field, 'エラーです')
    })

    test('表示', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('入力', () => {
    const newValue = '2020-09-22'
    beforeEach(async () => {
      const inputWrapper = getInputWrapper(wrapper)
      await inputWrapper.setValue(newValue)
    })

    test('MyForm.setValue()を呼出', () => {
      expect(form.setValue).toHaveBeenCalledWith(field, newValue)
    })
  })

  describe('クリアボタンクリック', () => {
    beforeEach(async () => {
      const buttonWrapper = getButtonWrapper(wrapper)
      await buttonWrapper.trigger('click')
    })

    test('MyForm.setValue()を呼出', () => {
      expect(form.setValue).toHaveBeenCalledWith(field, '')
    })
  })

  describe('オプション', () => {
    describe('showClear=false', () => {
      beforeEach(() => {
        wrapper.setProps({ showClear: false })
      })

      test('クリアボタン非表示', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
