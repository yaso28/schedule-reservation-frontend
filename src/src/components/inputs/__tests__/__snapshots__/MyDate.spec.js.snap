// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyDate オプション showClear=false クリアボタン非表示 1`] = `
<div
  class="form-row form-group"
>
  <label
    class="col-form-label col-sm-3"
    for="sample_field"
  >
    
    サンプル
  
  </label>
   
  <div
    class="col-sm-9"
  >
    <div
      class="my-input-wrapper"
    >
      <div
        class="d-flex"
      >
        <input
          class="form-control flex-grow-1"
          id="sample_field"
          type="date"
        />
         
        <!---->
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;

exports[`MyDate バリデーションエラー 表示 1`] = `
<div
  class="form-row form-group"
>
  <label
    class="col-form-label col-sm-3"
    for="sample_field"
  >
    
    サンプル
  
  </label>
   
  <div
    class="col-sm-9"
  >
    <div
      class="my-input-wrapper is-invalid"
    >
      <div
        class="d-flex"
      >
        <input
          class="form-control flex-grow-1"
          id="sample_field"
          type="date"
        />
         
        <button
          class="btn"
          type="button"
        >
          <svg
            class="c-icon"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="ci-primary"
              d="M96,472a23.82,23.82,0,0,0,23.579,24H392.421A23.82,23.82,0,0,0,416,472V152H96Zm32-288H384V464H128Z"
              fill="var(--ci-primary-color, currentColor)"
            />
            <rect
              class="ci-primary"
              fill="var(--ci-primary-color, currentColor)"
              height="200"
              width="32"
              x="168"
              y="216"
            />
            <rect
              class="ci-primary"
              fill="var(--ci-primary-color, currentColor)"
              height="200"
              width="32"
              x="240"
              y="216"
            />
            <rect
              class="ci-primary"
              fill="var(--ci-primary-color, currentColor)"
              height="200"
              width="32"
              x="312"
              y="216"
            />
            <path
              class="ci-primary"
              d="M328,88V40c0-13.458-9.488-24-21.6-24H205.6C193.488,16,184,26.542,184,40V88H64v32H448V88ZM216,48h80V88H216Z"
              fill="var(--ci-primary-color, currentColor)"
            />
          </svg>
        </button>
      </div>
    </div>
     
    <div
      class="invalid-feedback"
    >
      
      エラーです
    
    </div>
  </div>
</div>
`;

exports[`MyDate 通常 表示 1`] = `
<div
  class="form-row form-group"
>
  <label
    class="col-form-label col-sm-3"
    for="sample_field"
  >
    
    サンプル
  
  </label>
   
  <div
    class="col-sm-9"
  >
    <div
      class="my-input-wrapper"
    >
      <div
        class="d-flex"
      >
        <input
          class="form-control flex-grow-1"
          id="sample_field"
          type="date"
        />
         
        <button
          class="btn"
          type="button"
        >
          <svg
            class="c-icon"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="ci-primary"
              d="M96,472a23.82,23.82,0,0,0,23.579,24H392.421A23.82,23.82,0,0,0,416,472V152H96Zm32-288H384V464H128Z"
              fill="var(--ci-primary-color, currentColor)"
            />
            <rect
              class="ci-primary"
              fill="var(--ci-primary-color, currentColor)"
              height="200"
              width="32"
              x="168"
              y="216"
            />
            <rect
              class="ci-primary"
              fill="var(--ci-primary-color, currentColor)"
              height="200"
              width="32"
              x="240"
              y="216"
            />
            <rect
              class="ci-primary"
              fill="var(--ci-primary-color, currentColor)"
              height="200"
              width="32"
              x="312"
              y="216"
            />
            <path
              class="ci-primary"
              d="M328,88V40c0-13.458-9.488-24-21.6-24H205.6C193.488,16,184,26.542,184,40V88H64v32H448V88ZM216,48h80V88H216Z"
              fill="var(--ci-primary-color, currentColor)"
            />
          </svg>
        </button>
      </div>
    </div>
     
    <!---->
  </div>
</div>
`;
